analyzers/src/SonarAnalyzer.CSharp/Rules/ClearTextProtocolsAreSensitive.cs

```diff
@@ -0,0 +1,167 @@
+
+        protected override void Initialize(SonarAnalysisContext context) =>
+            context.RegisterCompilationStartAction(ccc =>
+            {
+                if (!IsEnabled(ccc.Options))
+                {
+                    return;
+                }
+
+                context.RegisterSyntaxNodeActionInNonGenerated(VisitStringExpressions, SyntaxKind.StringLiteralExpression, SyntaxKind.InterpolatedStringExpression);
+                context.RegisterSyntaxNodeActionInNonGenerated(VisitObjectCreation, SyntaxKind.ObjectCreationExpression);
+                context.RegisterSyntaxNodeActionInNonGenerated(VisitInvocationExpression, SyntaxKind.InvocationExpression);
+                context.RegisterSyntaxNodeActionInNonGenerated(VisitAssignments, SyntaxKind.SimpleAssignmentExpression);
+            });
+
+        private void VisitObjectCreation(SyntaxNodeAnalysisContext context)
+        {
+            var objectCreation = (ObjectCreationExpressionSyntax)context.Node;
+            if (!IsServerSafe(objectCreation) && objectInitializationTracker.ShouldBeReported(objectCreation, context.SemanticModel))
+            {
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(EnableSslRule, objectCreation.GetLocation()));
+            }
+
+            if (Regex.IsMatch(objectCreation.Type.ToString(), TelnetPatternForIdentifier))
+            {
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(DefaultRule, objectCreation.GetLocation(), Telnet, recommendedProtocols[Telnet]));
+            }
+        }
+
+        private void VisitInvocationExpression(SyntaxNodeAnalysisContext context)
+        {
+            var invocation = (InvocationExpressionSyntax)context.Node;
+            if (Regex.IsMatch(invocation.Expression.ToString(), TelnetPatternForIdentifier))
+            {
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(DefaultRule, invocation.GetLocation(), Telnet, recommendedProtocols[Telnet]));
+            }
+        }
+
+        private void VisitAssignments(SyntaxNodeAnalysisContext context)
+        {
+            var assignment = (AssignmentExpressionSyntax)context.Node;
+            if (assignment.Left is MemberAccessExpressionSyntax memberAccess
+                && memberAccess.IsMemberAccessOnKnownType(EnableSslName, KnownType.System_Net_FtpWebRequest, context.SemanticModel)
+                && context.SemanticModel.GetConstantValue(assignment.Right) is { HasValue: true, Value: bool enableSslValue }
+                && !enableSslValue)
+            {
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(EnableSslRule, assignment.GetLocation()));
+            }
+        }
+
+        private void VisitStringExpressions(SyntaxNodeAnalysisContext c)
+        {
+            var text = GetText(c.Node);
+            if (TryGetUnsafeProtocol(text, out var unsafeProtocol))
```

Comment:
Following the suggestion below for `TryGetUnsafeProtocol`, this could be updated afterwards
```suggestion
            if (GetUnsafeProtocol(GetText(c.Node)) is {} unsafeProtocol)
```